File: wallet-domain/src/main/java/com/example/wallet/domain/refund/RefundStatus.java
package com.example.wallet.domain.refund;

public enum RefundStatus {
    REQUESTED,   // 환불 요청만 등록
    APPROVED,    // PG 환불 승인
    COMPLETED,   // 사용자 Wallet 환원 완료
    FAILED;

    public boolean isTerminal() { return this == COMPLETED || this == FAILED; }
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/refund/RefundStatus.kt
package com.example.billing.domain.refund

enum class RefundStatus {
    REQUESTED,   // 환불 요청만 등록
    APPROVED,    // PG 환불 승인
    COMPLETED,   // 사용자 Wallet 환원 완료
    FAILED;

    fun isTerminal(): Boolean = this == COMPLETED || this == FAILED
}
